### PR TITLE
Ports TG Sofa + Bed changes from #62169 and #62813

### PR DESCRIPTION
**Pull request by KathrinBailey**
## About The Pull Request
• Ports https://github.com/tgstation/tgstation/pull/62169.
• Ports https://github.com/tgstation/tgstation/pull/62813
Ported from TG, which ported parts from SR.

## Why It's Good For The Game
More furniture variety for mappers and double beds so there's space for lazy slobs to put their pizza whilst they eat it in bed whilst watching movies.

## Changelog
:cl:
fix: Corporate sofa sprites are no longer wonky.
add: New sofa types from TG.
add: Double beds from TG/SR.
/:cl:

### CODE REVIEW

**Review by MissFox0810 — DISMISSED** [review]
Please make a seprate PR porting the benches as i know a few people would like that.
Will accept this Icon change for the sofas to also include the bench icons.

**Comment by KathrinBailey** [comment]
I'm so so sorry
image
I don't dismiss it, Github is lying.
I treasure it. I'm not as ungrateful as this is making me sound.

**Comment by Azarak** [comment]
This PR removes the ability of the linen bins to be able to handle towels aswell as a linen bin subtype that starts with towels

**Comment by MissFox0810** [comment]
Changing to draft untill this is ready for review

**Comment by MissFox0810** [comment]
Adding as a TM due to some core code changes. and potential better solutions.

**Comment by atakiya** [comment]
image
i did no such thing >:(

File: code/game/objects/structures/bedsheet_bin.dm [hunk 1]
```diff
@@ -406,13 +609,13 @@ LINEN BINS
...
 
-		var/obj/item/B
+		var/obj/item/bedsheet/B
 		if(sheets.len > 0)
 			B = sheets[sheets.len]
 			sheets.Remove(B)
 
 		else
-			B = new spawned_sheet(loc)
+			B = new /obj/item/bedsheet(loc)
```
Comment (Azarak) [inline code/game/objects/structures/bedsheet_bin.dm:618]:
```suggestion
			B = new spawned_sheet(loc)
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 2]
```diff
@@ -406,13 +609,13 @@ LINEN BINS
 	if(amount >= 1)
 		amount--
 
-		var/obj/item/B
+		var/obj/item/bedsheet/B
```
Comment (Azarak) [inline code/game/objects/structures/bedsheet_bin.dm:612]:
```suggestion
		var/obj/item/B
```

**Review by Azarak — APPROVED** [review]

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 3]
```diff
@@ -229,3 +229,23 @@
 /obj/structure/bed/maint/Initialize()
 	. = ..()
 	AddElement(/datum/element/swabable, CELL_LINE_TABLE_MOLD, CELL_VIRUS_TABLE_GENERIC, rand(2,4), 25)
+
+//Double Beds, for luxurious sleeping, i.e. the captain and maybe heads- if people use this for ERP, send them to skyrat
```
Comment (atakiya) [inline code/game/objects/structures/beds_chairs/bed.dm:233]:
```suggestion
```

File: code/game/objects/structures/beds_chairs/sofa.dm [hunk 4]
```diff
@@ -55,3 +55,26 @@
 
 /obj/structure/chair/sofa/corp/corner
 	icon_state = "corp_sofacorner"
+
+// Ported from Skyrat
```
Comment (atakiya) [inline code/game/objects/structures/beds_chairs/sofa.dm:59]:
```suggestion
```

**Review by atakiya — CHANGES_REQUESTED** [review]
Useless comments are not needed.

**Comment by Azarak** [comment]
Oh I didn't even notice those comments, yeah remove them

**Review by atakiya — APPROVED** [review]
Good from my side.

**Pull request merged**